kpatch-build/kpatch-elf.c

```diff
@@ -76,6 +76,31 @@ int is_debug_section(struct section *sec)
 	       !strncmp(name, ".eh_frame", 9);
 }
 
+#ifdef __powerpc__
+/*
+ * Number of instructions between global and local entry point of a
+ * function is mostly 8 instructions. i.e st_other == 3. The count of
+ * instruction can be decoded only for local function symbols.
+ */
+unsigned int has_local_entry_offset(GElf_Sym sym)
+{
+	return !!PPC64_LOCAL_ENTRY_OFFSET(sym.st_other);
+}
+
+/*
+ * As per ELF ABI v2 specification, any function those symbol reference
+ * is to be dynamically resolved. The st_shndx member of such symbol
+ * is set to SHN_UNDEF.
+ */
+unsigned int is_local_func(struct symbol *sym)
+{
+	if (sym->type != STT_FUNC || sym->sym.st_shndx == SHN_UNDEF)
+		return 0;
+
+	return 1;
```

Comment:
Consider the following case there the ``` sym->bind == STB_GLOBAL and sym->sym.st_shndx != SHN_UNDEF ```
```
51: 0000000000000008  1816 FUNC    LOCAL  DEFAULT [<localentry>: 8]     9 mmap_region
60: 0000000000000008    72 FUNC    GLOBAL DEFAULT [<localentry>: 8]     7 cleanup_module
65: 0000000000000008   916 FUNC    GLOBAL DEFAULT [<localentry>: 8]     5 init_module
```
just checking for ```sym->bind != STB_GLOBAL ```, will end in false positive.